WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApllicationLogDetailView.swift

```diff
@@ -0,0 +1,103 @@
+import SwiftUI
+
+/// Hosting controller that wraps an `ApplicationLogDetailView`
+///
+final class ApplicationLogDetailViewController: UIHostingController<ApplicationLogDetailView> {
+    init(with contents: String, for date: String) {
+        let viewModel = ApplicationLogViewModel(logText: contents)
+        super.init(rootView: ApplicationLogDetailView(viewModel: viewModel))
+        self.title = date
+        viewModel.present = { [weak self] vc in
+            self?.present(vc, animated: true, completion: nil)
+        }
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+struct ApplicationLogDetailView: View {
+    @ObservedObject var viewModel: ApplicationLogViewModel
+
+    private let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateStyle = .medium
+        formatter.timeStyle = .medium
+        return formatter
+    }()
+
+    var body: some View {
+        ScrollViewReader { scrollProxy in
+            List(viewModel.lines) { line in
+                VStack(alignment: .leading, spacing: 6) {
+                    line.date
+                        .flatMap(dateFormatter.string(for:))
```

Comment:
⚠️  What's the reason for having this logic here instead of on the view model?
If we move it there we could have the `dateFormatter` there where it won't get recreated, right?

Reply:
I agree that's better, moved it in d2eaf9839